docs/keyshare-protocol.md

```diff
@@ -79,29 +83,33 @@ In this way, the issuer enforces that the user uses the help of the keyshare ser
 
 ### Registration
 
-When registering, the IRMA app POSTs a message like the one below to the to `/api/v1/client/register` at the keyshare server:
+When registering, the IRMA app signs the following message into a JWT with its ECDSA private key, and sends that to `/client/register` at the keyshare server:
 
 ```json
 {
     "email": "example@example.com",
     "language": "en",
     "pin": "0kO3xbCrWMK1336eKzI3KOKWWogGb/oW4xErUd5rwFI=\n",
+    "publickey": "User's ECDSA public key, base64-encoded"
 }
 ```
 
-The email address is optional and may be absent. The `language` indicates the user's preferred language, used for a confirmation mail if the email address is present. Lastly, the `pin` field is computed as `Base64(SHA256(salt, pin))\n` (the trailing newline is there for legacy purposes and will be removed in the future).
+The email address is optional and may be absent. The `language` indicates the user's preferred language, used for a confirmation mail if the email address is present. Lastly, the `pin` field is computed as `Base64(SHA256(salt, pin))\n` (the trailing newline is there for legacy purposes and will be removed in the future). Note that since the user is not yet known to the keyshare server, this message is self-signed; to validate it the public key will first need to be extracted from the JWT and parsed.
 
 ### Authentication
 
-During an IRMA session, authenticating to the keyshare server during the protocol between the IRMA client and keyshare server is done as follows. After computing the PIN as `Base64(SHA256(salt, pin))\n`, a message like the following is sent to the keyshare server at `POST /api/v1/user/verify/pin`:
+During an IRMA session, authenticating to the keyshare server during the protocol between the IRMA client and keyshare server is done as follows. First, the app retrieves a challenge from the keyshare server at `POST /users/verify_start`. Next it computes the PIN as `Base64(SHA256(salt, pin))\n`, and using its ECDSA private key it signs the following message into a JWT, and sends that to the keyshare server at `POST /user/verify/pin_challengeresponse`:
```

Comment:
I guess the `\n` in `Base64(SHA256(salt, pin))\n` is by accident?

Reply:
No, see the remark about that [here](https://irma.app/docs/keyshare-protocol/#registration).